### PR TITLE
Request to join the Feature branch to the Main branch as part of performing the Bubble Sort algorithm relative to the first Microverse Ruby project.

### PR DESCRIPTION
**Pull request by aristides1000**
Request to join the Feature branch in the Main branch as part of the realization of the Bubble Sort algorithm relative to the first Microverse Ruby project, In this project I learned the importance of element ordering algorithms in arrays, in this case In particular I made use of the Bubble Sort methodology

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,70 @@
...
+Computer running [Mac OS](https://www.apple.com/macos/big-sur/), [Windows](https://www.microsoft.com/en-us/software-download/windows10), or [Linux Ubuntu](https://ubuntu.com/download). [Ruby V.3.0.0](https://www.ruby-lang.org/es/downloads/)...
+
+### Install
+
+Getting Started
+To get a local copy up and running follow these simple example steps:
+
+- You need to [clone](https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/cloning-a-repository) or [download as zip file](https://www.itprotoday.com/mobile-management-and-security/how-do-i-download-files-github) the repository on your device.
+- [Unzip](http://www.e7z.org/open-zip.htm) the repository or just clone it using git or github.
+- [Grab the index.html file and open it on your favorite browser.](https://helpdeskgeek.com/how-to/open-an-html-file-in-google-chrome/)
```
Comment (enaburekhan) [inline README.md]: Kindly correct this line to reflect the current project which is `bubble_sort `using the `ruby` language.

File: bubble_sort.rb [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+      j += 1
+    end
+    i += 1
+  end
+  array
+end
+
+p bubble_sort_by(%w[hi hello hey]) do |left, right|
+  left.length - right.length
+end
```
Comment (enaburekhan) [inline bubble_sort.rb]: bubble_sort_by raises an error if block is not given. expected `LocalJumpError` but nothing was raised. You need to refactor the method to use `yield`.

**Review by enaburekhan — CHANGES_REQUESTED** [review]
# Status: Changes required:full_moon:.
**Hi Molina!
Great job on your first ruby project:clap::clap:.**

**Major Highlights**

- The PR and readme is very descriptive
- The codebase is clean and the written methods are easy to read and understand.

However, we just need to make a few changes to get the project approved:muscle:.

**To improve**
- [x] As highlighted the PR below. Refactor the bubble_sort_by method to use yield. And pay attention to what's been yielded.  the test is failing because `bubble_sort_by is suppose to raises an error if block is not given`.  A local jump error was expected, but nothing was raised. Kindly fix this.

- [x] Also, correct the highlighted line 34 in the readme to reflect the current project. Which is `bubble_sort in ruby`.

Make the above-required changes and when you are ready, request another review.
Do have questions/comments? kindly drop them :point_down: and ping me on slack @ Eric Enaburekhan to respond.
Happy hacking:blush:
Eric Enaburekhan

**Comment by aristides1000** [comment]
Thank you very much for the corrections, I fixed my bubble_sort.rb file and my README.md file, according to the indications given in the previous correction.

**Review by DeliceLydia — APPROVED** [review]
# Status: Approved image

Congratulations @aristides1000 for implementing changes requested by the previous TSE and this project shows that you really worked hard and your bubble_sort is running perfectly now

## Kudos once again and feel free to merge and move on 

N.B: next time try to make a title of your PR short and descriptive

All the best,

Lydia Ingabire

**Pull request merged**